tests/lib/fakeinstaller/main.go

```diff
@@ -35,9 +38,71 @@ import (
 	"github.com/snapcore/snapd/osutil"
 	"github.com/snapcore/snapd/osutil/disks"
 	"github.com/snapcore/snapd/osutil/mkfs"
-	"github.com/snapcore/snapd/overlord/state"
 )
 
+func waitForDevice() string {
+	for {
+		devices, err := emptyFixedBlockDevices()
+		if err != nil {
+			logger.Noticef("cannot list devices: %v", err)
+		}
+		switch len(devices) {
+		case 0:
+			logger.Noticef("cannot use automatic mode, no empty disk found")
+		case 1:
+			// found exactly one target
+			return devices[0]
+		default:
+			logger.Noticef("cannot use automatic mode, multiple empty disks found: %v", devices)
+		}
+		time.Sleep(5 * time.Second)
+	}
+}
+
+// emptyFixedBlockDevices finds any non-removalble physical disk that has
+// no partitions. It will exclude loop devices.
+func emptyFixedBlockDevices() (devices []string, err error) {
+	// eg. /sys/block/sda/removable
+	removable, err := filepath.Glob(filepath.Join(dirs.GlobalRootDir, "/sys/block/*/removable"))
+	if err != nil {
+		return nil, err
+	}
+	for _, removableAttr := range removable {
+		val, err := ioutil.ReadFile(removableAttr)
+		if err != nil || string(val) != "0\n" {
+			// removable, ignore
+			continue
+		}
+		// let's see if it has partitions
+		dev := filepath.Base(filepath.Dir(removableAttr))
+		if strings.HasPrefix(dev, "loop") {
+			// is loop device, ignore
+			continue
+		}
+		pattern := fmt.Sprintf(filepath.Join(dirs.GlobalRootDir, "/sys/block/%s/%s*/partition"), dev, dev)
+		// eg. /sys/block/sda/sda1/partition
+		partitionAttrs, _ := filepath.Glob(pattern)
+		if len(partitionAttrs) != 0 {
+			// has partitions, ignore
+			continue
+		}
+		devNode := fmt.Sprintf("/dev/%s", dev)
+		output, err := exec.Command("lsblk", "--output", "fstype", "--json", devNode).CombinedOutput()
+		if err != nil {
+			return nil, osutil.OutputErr(output, err)
+		}
+		// TODO: parser proper json
```

Comment:
I would remove this command, and maybe instead write:

> No need to do proper JSON parsing here, since we only care whether the output is like <show example output>

Though I wonder if it wouldn't be better to avoid using `strings.Contains` and have a `==` match on `output.Trim()` instead, below.

Reply:
In the end I removed the json option and just used output without headers, using lsblk this way from a program should be fine and keep compatibility with future versions.